src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,107 @@
-import { useState } from 'react';
+import { FormEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isDisabled = () => {
+    return title === ''
+      || imgUrl === ''
+      || imdbUrl === ''
+      || imdbId === '';
+  };
+
+  const newMovie: Movie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handleSubmitForm = (event: FormEvent) => {
+    event.preventDefault();
+
+    onAdd(newMovie);
+    setCount(prev => prev + 1);
+    resetForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event: FormEvent) => handleSubmitForm(event)}
```

Comment:
No need to add wrapper here, if you use event as argument.
```suggestion
      onSubmit={handleSubmitForm}
```